### DISCUSSION TITLE
UnexpectedEof when calling AsyncBufReadExt.read_until

### DISCUSSION THREAD

**Discussion by lufte** [tokio-rs/tokio#5615] [Q&A] [answered]
According to the [documentation](https://docs.rs/tokio/latest/tokio/io/trait.AsyncBufReadExt.html#method.read_until), `read_until` should read all bytes until the delimiter byte or EOF is found. In one particular scenario I'm getting `UnexpectedEof` when using it and I don't understand what this means. `read_until` is not decoding the bytes into UTF-8 or anything like that, so no EOF should be "unexpected". I appreciate any help.

**Comment by Darksonn** [comment, marked as answer]
According to the source code of `read_until`, it never creates `UnexpectedEof` errors. Perhaps the error came from the underlying IO resource that you're calling `read_until` on? Alternatively, are you sure that the error came from `read_until`? If you also have a `read_exact` call somewhere, then that could be the cause.

**Comment by lufte** [comment]
I'm sure the error came from `read_until`, so I'll try to investigate if it's coming from the object wrapped by my `BufReader`, which is a `tokio_rustls::client::TlsStream`. Thanks, it didn't occur to me it could be a propagated error.
